src/PlotlyEditor.js

```diff
@@ -1,280 +1,67 @@
-        graphDiv.data.push({type: 'scatter', mode: 'markers'});
-        if (this.props.afterAddTrace) {
-          this.props.afterAddTrace(payload);
-        }
-        if (this.props.onUpdate) {
-          this.props.onUpdate();
-        }
-        break;
-
-      case EDITOR_ACTIONS.DELETE_TRACE:
-        if (payload.traceIndexes && payload.traceIndexes.length) {
-          if (this.props.beforeDeleteTrace) {
-            this.props.beforeDeleteTrace(payload);
-          }
-          graphDiv.data.splice(payload.traceIndexes[0], 1);
-          if (this.props.afterDeleteTrace) {
-            this.props.afterDeleteTrace(payload);
-          }
-          if (this.props.onUpdate) {
-            this.props.onUpdate();
-          }
-        }
-        break;
-
-      case EDITOR_ACTIONS.DELETE_ANNOTATION:
-        if (isNumeric(payload.annotationIndex)) {
-          if (this.props.beforeDeleteAnnotation) {
-            this.props.beforeDeleteAnnotation(payload);
-          }
-          graphDiv.layout.annotations.splice(payload.annotationIndex, 1);
-          if (this.props.afterDeleteAnnotation) {
-            this.props.afterDeleteAnnotation(payload);
-          }
-          if (this.props.onUpdate) {
-            this.props.onUpdate();
-          }
-        }
-        break;
-
-      case EDITOR_ACTIONS.DELETE_SHAPE:
-        if (isNumeric(payload.shapeIndex)) {
-          if (this.props.beforeDeleteShape) {
-            this.props.beforeDeleteShape(payload);
-          }
-          graphDiv.layout.shapes.splice(payload.shapeIndex, 1);
-          if (this.props.afterDeleteShape) {
-            this.props.afterDeleteShape(payload);
-          }
-          if (this.props.onUpdate) {
-            this.props.onUpdate();
-          }
-        }
-        break;
-
-      case EDITOR_ACTIONS.DELETE_IMAGE:
-        if (isNumeric(payload.imageIndex)) {
-          if (this.props.beforeDeleteImage) {
-            this.props.beforeDeleteImage(payload);
-          }
-          graphDiv.layout.images.splice(payload.imageIndex, 1);
-          if (this.props.afterDeleteImage) {
-            this.props.afterDeleteImage(payload);
-          }
-          if (this.props.onUpdate) {
-            this.props.onUpdate();
-          }
-        }
-        break;
-
-      default:
-        throw new Error('must specify an action type to handleEditorUpdate');
-    }
+  constructor(props) {
+    super();
+    this.state = {graphDiv: {}};
+    this.PlotComponent = createPlotComponent(props.plotly);
   }
 
   render() {
     return (
-      <div
-        className={
-          bem('plotly-editor') +
-          ' plotly-editor--theme-provider' +
-          `${this.props.className ? ` ${this.props.className}` : ''}`
-        }
-      >
-        <ModalProvider>
-          {this.props.graphDiv &&
-            this.props.graphDiv._fullLayout &&
-            (this.props.children ? this.props.children : <DefaultEditor />)}
-        </ModalProvider>
+      <div className="app">
```

Comment:
Should `app` here be replaced with a BEM class instead? (Perhaps by adjusting/moving the one now in `EditorControls`?)

Reply:
Good point, I just copied stuff over but `app` is a terrible class to bake in ;)